Issue #3809: Add python client for Roles and Policies

schlameel opened the pull request:
Includes Tests

### Describe your changes :
<!-- Explain what you have done & tag your assigned issue !-->
Add Python client for Roles and Policies

- PATCH the Policies for a Role
- PATCH the Rules for a Policy
- complete tests for Roles and Policies including new functionality

#
### Type of change :
<!-- You should choose 1 option and delete options that aren't relevant -->
- [ ] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
N/A

#
### Checklist:
<!-- add an x in [] if done, don't mark items that you didn't do !-->
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
<!-- Please see the contributing guidelines and then add your reviewer(s) !-->
<!--- OpenMetadata community thanks you for explaining your changes in detail !-->
<!--- If you are unsure of people to review your work, you can add anyone of these developers :) !-->
<!--- Frontend: @open-metadata/ui -->
<!--- Backend: @open-metadata/backend -->
<!--- Ingestion: @open-metadata/ingestion -->

pmbrull commented on ingestion/src/metadata/ingestion/ometa/mixins/role_policy_mixin.py:
```diff
@@ -0,0 +1,440 @@
+VALUE_RESOURCES: str = "resources"
+
+# Operations
+ADD = "add"
```
should we centralize these variables that we are using both here and in patch_mixin.py?

pmbrull replied:
hi, thanks for the ping. The enums I might add them to https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/ometa/client_utils.py

pmbrull commented on ingestion/src/metadata/ingestion/ometa/mixins/role_policy_mixin.py:
```diff
@@ -0,0 +1,440 @@
+        :param previous: the field from the previous rule
+        :param current: the field from the current rule
+        :param rule_index: the index of the previous rule
+        :param path: path string for the filed
```
nit - Sometimes we use `:param ....:` and sometimes `Args...` for the docstring. Could we keep it all as `Args` please?

pmbrull commented on ingestion/src/metadata/ingestion/ometa/mixins/role_policy_mixin.py:
```diff
@@ -0,0 +1,440 @@
+REMOVE = "remove"
+
+# Paths
+ROLES_PATH: str = "/roles/{role_id}"
```
Thinking if it might be easier to read and use it if we just wrap all these constants in an `Enum`?

pmbrull replied:
We can name the Enums as

```python
class PatchField(str, Enum):
    OPERATION = "op"
    PATH = "path"
    VALUE = "value"
```

pmbrull reviewed (commented):
Thanks for the effort here. Left a few nit / styling comments if you have the time

schlameel commented:
- Moved constants to enums in `/ingestion/src/metadata/ingestion/ometa/patch.py`
- Merged latest commit
- Updated constants for merged `patch_mixin.py` and updated new methods in `patch_mixin.py` to use the enums.
- Fixed requested "nits"

pmbrull reviewed (approved):
Thanks for the changes! LGTM!
